clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -168,6 +169,11 @@ public SSLEngine createSslEngine(String peerHost, int peerPort) {
         return sslEngineBuilder.createSslEngine(mode, peerHost, peerPort, endpointIdentification);
     }
 
+    // Keep this for backward compatibility
```

Comment:
Maybe say something similar to the PR about this method being commonly used to set up ssl clients even though not public API so we're temporarily keeping it for backwards compatibility as of 2.3 (that will give a sense of timing when people see this code a year or two from now).

Reply:
Another option would be to put a "deprecated" annotation here.

Reply:
That's a good point.

Reply:
Good suggestion. I changed the comment to a deprecated annotation.